Fatras/include/ActsFatras/Kernel/detail/SimulationActor.hpp

```diff
@@ -186,6 +186,11 @@ struct SimulationActor {
           before.fourMomentum(), after.fourMomentum(), result.hits.size());
     }
 
+    if (after.absoluteMomentum() == 0.0) {
+      result.isAlive = false;
+      return;
+    }
```

Comment:
So this is only to inhibit the FPE, that comes from computing `q / p` with `p =  0` exactely in `updateStepper`.
Also not sure if the `Aborter` helps here, because I think it only jumps in after the call to the `Actor`, and `updateStepper` is called anyways.

But if you say there is this aborter, which terminates the proagator anyways, it might be better to do something like this instead:

```c++
if (after.absoluteMomentum() > 0.0) {
 stepper.update(state.stepping, after.position(), after.unitDirection(), after.absoluteMomentum(), after.time());
}
```

Because then just then the aborter will jump in afterwards. Is this correct?

Reply:
I think the aborter is not wired to the fatras simulation yet. you can find it here https://github.com/acts-project/acts/blob/main/Core/include/Acts/Propagator/StandardAborters.hpp#L193

what surprises me is that we call the stepper from the actor here. I dont understand why this is necessary

anyways for the sake of progress I think we can just merge it as is

Reply:
> what surprises me is that we call the stepper from the actor here. I dont understand why this is necessary

Hmm that is a common pattern, should be done similar in the CKF. Somehow you need to update the stepper state with your information computed in the `Actor`